provider/CONTRIBUTE.md

```diff
@@ -0,0 +1,140 @@
+
+### Go tools
+
+Terraform is developed using Go. You need to install Go **1.11.x** to be able to build and debug the provider locally.
+You can download it from [this page](https://golang.org/dl/) and find the installation instructions for your system [here](https://golang.org/doc/install#install)
+
+Then you can test your environment following the instructions on [this page](https://golang.org/doc/install#testing).
+
+### Check you GOPATH
+
+As of many Go project, AzureRM Terraform provider rely on your GOPATH environment variable. You may want to make sure it is well configured for your system, reading [this page](https://github.com/golang/go/wiki/SettingGOPATH).
+
+### Visual Studio Code
+
+You can use the IDE you love, but in this documentation we will describe how to contribute to the Terraform AzureRM provider using Visual Studio Code. You can download it for your system from [this page](https://code.visualstudio.com/Download).
+
+Once installed, download the Go extension for VS Code:
+
+![VS Code Go Extension](assets/ms-vscode-go.png)
+
+Once installed, open VS Code and look for the `Go: Install/Update Tools` in the command palette, a select all the tools:
+
+![Install Go Tools](assets/code-install-go-tools.png)
+
+![Install Go Tools - All](assets/code-install-go-tools-all.png)
+
+![Install Go Tools - Wait](assets/code-install-go-tools-wait.png)
+
+### Specific requirements for Windows users
+
+If you are running Windows, then you need to install Git Bash and Make for Windows. Check the dedicated section on Terraform on Azure repository [here](https://github.com/terraform-providers/terraform-provider-azurerm#windows-specific-requirements).
+
+## Get the sources
+
+First, go to the [AzureRM Terraform provider](https://github.com/terraform-providers/terraform-provider-azurerm) project page and fork the repository into your GitHub account.
+
+Once done, you need to clone your fork into the `$GOPATH/src/github.com/terraform-providers/terraform-provider-azurerm` folder.
+
+## Build the sources
+
+You can check that everything is OK by building the AzureRM provider:
+
+```bash
+cd $GOPATH/src/github.com/terraform-providers/terraform-provider-azurerm
+make build
+```
+
+Once completed, the binary of the AzureRM provider should be available in the `$GOPATH/bin` directory.
+
+*Note: on Windows, you need to use Git Bash*
+
+More information [here](https://github.com/terraform-providers/terraform-provider-azurerm#developing-the-provider).
+
+## Work with your local build
+
+Once you have built a new version of the AzureRM Terraform provider, you can use it locally.
+To use your local version, the first thing to do is a `terraform init`, as usual, to inialize your terraform working directory.
+
+The init operation will download the AzureRM Provider for you. You can just remove it, and replace it with your local copy.

```

Comment:
Shouldn't this use the local version as long as it is in your go path? In other words, init should not download the released version because it finds the one that was just built. @jeffreyCline please confirm

Reply:
@grayzu I've done some testings, but was not able to reproduce the behavior you describe. Even if I have the locally built azure rm provider in $GOPATH/bin and $GOPATH/bin in my path environment variable, the terraform init download the latest version of the provider. @jeffreyCline - if there is a simpler way, I'll be happy to get your inputs. thanks!

Reply:
Ok let's leave the it as is for now. We can optimize later if need be.